packages/fast-components-react-base/src/viewport-positioner/viewport-positioner.tsx

```diff
@@ -0,0 +1,1057 @@
+    private scrollLeft: number = 0;
+
+    /**
+     * constructor
+     */
+    constructor(props: ViewportPositionerProps) {
+        super(props);
+
+        this.state = {
+            disabled: true,
+            noOberverMode: false,
+            xTransformOrigin: "left",
+            yTransformOrigin: "top",
+            xTranslate: 0,
+            yTranslate: 0,
+            top: null,
+            right: null,
+            bottom: null,
+            left: null,
+            currentHorizontalPosition:
+                ViewportPositionerHorizontalPositionLabel.uncontrolled,
+            currentVerticalPosition: ViewportPositionerVerticalPositionLabel.uncontrolled,
+            defaultHorizontalPosition: this.ConvertHorizontalPositionToLabel(
+                this.props.horizontalPositioningMode,
+                this.props.defaultHorizontalPosition
+            ),
+            defaultVerticalPosition: this.ConvertVerticalPositionToLabel(
+                this.props.verticalPositioningMode,
+                this.props.defaultVerticalPosition
+            ),
+            initialLayoutComplete: false,
+        };
+    }
+
+    public componentDidMount(): void {
+        this.checkComponentConfig();
+        this.requestFrame();
+    }
+
+    public componentWillUnmount(): void {
+        this.disableComponent();
+    }
+
+    public componentDidUpdate(prevProps: ViewportPositionerProps): void {
+        if (
+            prevProps.viewport !== this.props.viewport ||
+            prevProps.anchor !== this.props.anchor ||
+            prevProps.disabled !== this.props.disabled
+        ) {
+            this.checkComponentConfig();
+        }
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div
+                ref={this.rootElement}
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+                style={this.getPositioningStyles()}
+            >
+                {this.props.children}
+            </div>
+        );
+    }
+
+    /**
+     * Create class-names
+     */
+    protected generateClassNames(): string {
+        let classNames: string = get(this.props, "managedClasses.viewportPositioner", "");
+
+        switch (this.state.currentHorizontalPosition) {
+            case ViewportPositionerHorizontalPositionLabel.left:
+                classNames = classNames.concat(
+                    " ",
+                    get(this.props.managedClasses, "viewportPositioner__left", "")
+                );
+                classNames = classNames.concat(
+                    " ",
+                    get(
+                        this.props.managedClasses,
+                        "viewportPositioner__horizontalFlipOutward",
+                        ""
+                    )
+                );
+                break;
+
+            case ViewportPositionerHorizontalPositionLabel.centerLeft:
```

Comment:
can we call this insetLeft, correct me if I'm wrong but centerLeft doesn't really tell me this is inset to the component

Reply:
yep